pkgs/jovian-greeter/greeter.py

```diff
@@ -122,11 +125,15 @@ def _recv(self) -> Mapping[str, Any]:
 class Context:
     def __init__(self, user: str, home: Path):
         self.user = user
+        logging.debug("USER: {}".format(self.user))
```

Comment:
This is probably unnecessary, you can inspect the environment via /proc or systemd or any other tool.

Reply:
Unless the script is dying after ~100ms. ;)

Reply:
In that case you should probably have a stacktrace?

Reply:
Not if it exited with `sys.exit(1)` due to misconfiguration. And even if you do, the stack trace won't always tell you what you need to know in my experience.

Reply:
Then the problem is that the script is doing `sys.exit(1)` without providing adequate information.

Reply:
No, because again, the stack trace won't always give you what you need -- unless your plan is to catch every exception in the entire program and then do your conditional dump.

The point of the logging system is usually to be as verbose as you could possibly need, and then quiet it by setting the log level higher than DEBUG/TRACE. Usually on WARN or something. Then provide a way to override the debug level if something goes wrong. Since this is a daemon configured by a Nix expression, maybe have the nix expression control a flag that in turn sets the debug level, if you're worried about cluttering up logs?

Reply:
Not to say that exiting with more information isn't a good idea, but dumping the entire execution context only iff the program exits is usually not done because there's a gradual accumulation of state information you need to dump as programs grow in size and complexity. Although this is a trivial Python script, I still think it would make sense to follow that pattern on account of the fact that Python is not a correctness-oriented language and whatever debugging mechanism you supply needs to account for it failing somewhere along the execution chain in a way you didn't expect.